src/rgw/rgw_sal_daos.cc

```diff
@@ -392,21 +392,9 @@ int DaosBucket::open(const DoutPrefixProvider* dpp) {
   }
 
   int ret;
-  daos_cont_info_t cont_info;
-  // TODO: We need to cache open container handles
-  ret = daos_cont_open(store->poh, info.bucket.name.c_str(), DAOS_COO_RW, &coh,
-                       &cont_info, nullptr);
-  ldpp_dout(dpp, 20) << "DEBUG: daos_cont_open, name=" << info.bucket.name
-                     << ", ret=" << ret << dendl;
-
-  if (ret != 0) {
-    return -ENOENT;
-  }
-
-  uuid_copy(cont_uuid, cont_info.ci_uuid);
-
-  ret = dfs_mount(store->poh, coh, O_RDWR, &dfs);
-  ldpp_dout(dpp, 20) << "DEBUG: dfs_mount ret=" << ret << dendl;
+  // How can i get the pool name?
+  const auto& daos_pool = g_conf().get_val<std::string>("daos_pool");
+  ret = dfs_connect(daos_pool.c_str(), nullptr, get_name().c_str(), O_RDWR, nullptr, &dfs);
 
   if (ret != 0) {
     daos_cont_close(coh, nullptr);
```

Comment:
This should not be needed, failure returns a closed handle.

Reply:
Also, maybe log an error/debug message to the log in failure, this is what `ldpp_dout(dpp, 20)` is for. 20 is debug, 0 is error.